### DISCUSSION TITLE
PredSim for Linux

### DISCUSSION THREAD

**Discussion by davidpagnon** [KULeuvenNeuromechanics/PredSim#163] [Feature requests & Suggestions]
**Is your feature request related to a problem? Please describe.**

No

**Describe the solution you'd like**

PredSim for Linux.

**Describe alternatives you've considered**

**Additional context**

**Are you willing to contribute in the development of this feature?**

No. I don't think I have the skills.

Dear PredSim community,
Thank you for releasing such a nice tool! I would like to test it, but I mostly run on Linux, and preferably on a server. 
Do you by any chance plan to support Linux?
Best regards

**Comment by davidpagnon** [comment]
And while I'm at it, is there any plan to make it run on Python?

I do know how much work it takes to convert such large projects to different languages and platforms, so I am really just asking in case it is already in the works.

**Comment by Lars-DHondt-KUL** [comment]
Hey, 

Sorry for the late reply, looks like I don't get notifications for new issues.

> I would like to test it, but I mostly run on Linux, and preferably on a server.
> Do you by any chance plan to support Linux?

There are no concrete plans to support Linux. We use a beefy desktop computer (windows) when we want to run a lot of simulations, but I can see that it would be convenient if simulations could also run on a server or supercomputer node that uses Linux.

The PredSim code has only one Windows-specific part (as far as I'm aware), and there are plans to change this part in the next few months. 

Part of the pre-processing relies on OpenSimAD ([submodule](https://github.com/Lars-DHondt-KUL/opensimAD/tree/144a09414a52ed4fa3c538580dc8d79ee5f56bfb)) to generate some files used by the simulation. Getting this to run on Linux will be more challenging. Worst case, we have to run pre-processing on a Windows machine and then transfer the files to the Linux server to run the actual simulation.

> And while I'm at it, is there any plan to make it run on Python?

No, but you could check this out: https://github.com/antoinefalisse/predsim_tutorial

**Comment by davidpagnon** [comment]
Thank you for your answer! I'm not currently on it right now, but I will probably get back to it and keep you updated.
